servicecontrol/troubleshooting.md

```diff
@@ -173,3 +175,27 @@ If many indexes are affected it may be easier to rebuild all indexes, although t
 - Navigate to the [database folder](configure-ravendb-location.md) on disk
 - Delete the `Indexes` folder
 - Start the ServiceControl instance
+
+
+## High CPU utilization
+
+Warn:  Do not terminate the process as  this can cause index corruption and cause the indexes to be verified of rebuils which causes long and exessive resource utilization for large datababases.
+
+If a ServiceControl instance shows high CPU utilization than this usually is caused due to:
+
+- Termination of the process and ServiceControl its internal database validating the indexes at the next startup.
+- Rebuilding of indexes which require all database context to be read and indexed.
+- Large ingestion of messages due to a huge backlog of messages in the queue.
+
+Usually this fixes itself over time as long as the process is not terminated. Do not terminate the process!
+
+Info: It is adviced to host ServiceControl instances on isolated (virtual) machine with dedicated (non shared) resources for it to not affect any other processes when utilization a lot of system resources.
+
+Resolution:
+
+- Ensure the latest version of ServiceControl is installed to ensure it has all latest performance enhancements. The most recent version is available at <https://particular.net/downloads>
+- Ensure storage disks are **at least** capable of 7,500 IOPS as stated in the [hardware considerations for ServiceControl](/servicecontrol/servicecontrol-instances/hardware.md). If the system is continually producing messages or generate lots or very large messages (several kilobytes or larges) that ServiceControl requires even faster disks and the **minimum** requirements.
+- Ensure no custom checks are shown in ServicePulse that are indicating Index issues. The log file could indicate the type of index issues (See [stale indexes](#stale-indexes), [index errors](#index-errors), and [corrupted indexes](#corrupted-indexes))
+- Consider disabling *Full Text Search* os message bodies and headers as this causes most resource utilization for CPU and disk IO. This can be disabled inrunning the latest version of ServiceControl by going to each ServiceControl instance type, opening the configuration (gear icon), scrolling down to Advanced Configuration and change Full Text Search On Message Bodies to Off and select Save and then restart the instance.
```

Comment:
```suggestion
- Consider disabling message bodies and headers *Full-Text search* as this causes most resource utilization for CPU and disk IO. This can be disabled in the latest version of ServiceControl by configuring each ServiceControl instance: open configuration (gear icon), scroll down to Advanced Configuration and set "Full-Text Search On Message Bodies" to Off, finally select Save, and then restart the instance.
```